contracts/interchain-token-service/src/executable.rs

```diff
@@ -7,14 +7,42 @@
 
 use soroban_sdk::{contractclient, Address, Bytes, BytesN, Env, String};
 
-/// Interface for an Interchain Token Executable app.
-#[contractclient(name = "InterchainTokenExecutableClient")]
-pub trait InterchainTokenExecutableInterface {
-    /// Return the trusted interchain token service contract address.
+/// This trait must be implemented by a contract to be compatible with the [`InterchainTokenExecutableInterface`].
+///
+/// To make a contract executable by the interchain token service contract, it must implement the [`InterchainTokenExecutableInterface`] trait.
+/// For security purposes and convenience, sender authorization and other commonly shared code necessary to implement that trait can be automatically generated with the [`axelar_soroban_std::Executable`] derive macro.
+/// All parts that are specific to an individual ITS executable contract are collected in this [`CustomExecutable`] trait and must be implemented by the contract to be compatible with the [`InterchainTokenExecutableInterface`] trait.
+///
+/// Do not add the implementation of [`CustomExecutable`] to the public interface of the contract, i.e. do not annotate the `impl` block with `#[contractimpl]`
+pub trait CustomExecutable {
+    /// The type of error the [`CustomExecutable::execute`] function returns. Generally matches the error type of the whole contract.
+    type Error: Into<soroban_sdk::Error>;
+
+    /// Returns the address of the interchain token service contract that is authorized to execute arbitrary payloads on this contract
     fn interchain_token_service(env: &Env) -> Address;
 
+    /// The custom execution logic that takes in an arbitrary payload and a token.
+    /// At the time this function is called, the calling address has already been verified as the correct interchain token service contract.
+    fn execute(
+        env: &Env,
+        source_chain: String,
+        message_id: String,
+        source_address: Bytes,
+        payload: Bytes,
+        token_id: BytesN<32>,
+        token_address: Address,
+        amount: i128,
+    ) -> Result<(), Self::Error>;
+}
+
+/// Interface for an Interchain Token Executable app. Use the [axelar_soroban_std::Executable] derive macro to implement this interface.
+///
+/// DO NOT IMPLEMENT THIS MANUALLY!
```

Comment:
Could you maybe emphasize this comment like this if it's important?
```suggestion
/// ***********************************************************************
/// * INTERFACE FOR AN INTERCHAIN TOKEN EXECUTABLE APP                   *
/// *                                                                    *
/// * Use the [axelar_soroban_std::Executable] derive macro to           *
/// * implement this interface.                                          *
/// *                                                                    *
/// * >>>>> DO NOT IMPLEMENT THIS MANUALLY! <<<<<                        *
/// ***********************************************************************
```

Reply:
That get's rendered in an unreadable manner as a doctest, but I made the last line bold now